src/multisig/multisigTransactionsOutcomeParser.ts

```diff
@@ -0,0 +1,278 @@
+import { TransactionOnNetwork } from "../core";
+import { Address } from "../core/address";
+import { Err } from "../core/errors";
+import { TransactionEvent } from "../core/transactionEvents";
+import { SmartContractCallOutcome, SmartContractResult } from "../transactionsOutcomeParsers/resources";
+
+enum Events {
+    MultisigDeploy = "MultisigDeploy",
+    MultisigActionProp = "MultisigActionProposal",
+    SignalError = "signalError",
+    WriteLog = "writeLog",
+}
+
+/**
+ * Parses the outcome of multisig contract operations
+ */
+export class MultisigTransactionsOutcomeParser {
+    /**
+     * Parses the outcome of creating a new multisig contract
+     * @param transactionOnNetwork The completed transaction
+     * @returns An array of objects containing the new contract addresses
+     */
+    parseDeployMultisigContract(transactionOnNetwork: TransactionOnNetwork): { contractAddress: Address }[] {
+        const directCallOutcome = this.findDirectMultisigDeployOutcome(transactionOnNetwork);
+
+        if (!directCallOutcome || directCallOutcome.returnCode !== "ok") {
+            return [];
+        }
+
+        // Look for the deployment events
+        const events = transactionOnNetwork.logs.events
+            .concat(transactionOnNetwork.smartContractResults.flatMap((result) => result.logs.events))
+            .filter((event) => event.identifier === Events.MultisigDeploy);
+
+        return events.map((event) => {
+            // Assuming the contract address is in the first topic
+            const addressBytes = Buffer.from(event.topics[0]);
+            const address = Address.newFromHex(addressBytes.toString("hex"));
+            return { contractAddress: address };
+        });
+    }
+
+    /**
+     * Parses the outcome of a multisig action proposal
+     * @param transactionOnNetwork The completed transaction
+     * @returns The action ID that was created
+     */
+    parseActionProposal(transactionOnNetwork: TransactionOnNetwork): number {
+        const directCallOutcome = this.findDirectMultisigCallOutcome(transactionOnNetwork);
+
+        if (!directCallOutcome || directCallOutcome.returnCode !== "ok") {
+            throw new Err("Failed to propose action: " + directCallOutcome.returnMessage);
+        }
+
+        if (directCallOutcome.returnDataParts.length === 0) {
+            throw new Err("No action ID returned in the transaction outcome");
+        }
+
+        // Assuming the first return data part contains the action ID as bytes
+        const actionIdBytes = directCallOutcome.returnDataParts[0];
+        return parseInt(Buffer.from(actionIdBytes).toString("hex"), 16);
```

Comment:
Can stay as it is, or use something from here, if something useful is available (and, if not available, should we add something there?):
 - https://github.com/multiversx/mx-sdk-js-core/blob/main/src/core/utils.codec.ts
 - https://github.com/multiversx/mx-sdk-js-core/blob/main/src/abi/codec/utils.ts

Or maybe we should / can directly use `NumericalBinaryCodec` - just an idea, maybe doesn't fit well.